lib/vmdb/config.rb

```diff
@@ -7,19 +7,23 @@ class Config
 
     require_relative 'configuration_encoder'
 
-    def self.clone_auth_for_log(auth)
-      log_auth = auth.deep_clone
-      [:bind_pwd, :amazon_secret].each do |key|
-        log_auth[key] = '********' if log_auth.key?(key)
-        if log_auth.key?(:user_proxies)
-          log_auth[:user_proxies].each do |p|
-            p[key] = '********' if p.key?(key)
-          end
-        end
+    def self.deep_clean_auth_for_log(log_auth)
+      log_auth.each do |key, val|
+        log_auth[key] = deep_clean_auth_for_log(val) if val.kind_of?(Hash)
+
+        log_auth[key] = "********" if Vmdb::ConfigurationEncoder::PASSWORD_FIELDS.include?(key.to_s)
+
+        log_auth[key].each do |p|
+          p.keys.each { |key2| p[key2] = "********" if Vmdb::ConfigurationEncoder::PASSWORD_FIELDS.include?(key2.to_s) }
+        end if [:user_proxies].include? key
       end
       log_auth
```

Comment:
@durandom Thank you for the suggestion but I think each solution has equal merits and this is more of a style decision so I would prefer leaving it as it stans.